examples/rust/pipe-ping-pong.rs

```diff
@@ -168,109 +168,164 @@ fn close(libos: &mut LibOS, pipeqd: QDesc) {
+        self.rx_pipeqd = match self.libos.create_pipe(&format!("{}:rx", pipe_name)) {
+            Ok(qd) => Some(qd),
+            Err(e) => anyhow::bail!("failed to open memory queue: {:?}", e),
+        };
 
-    // Setup peer.
-    let pipeqd_tx: QDesc = match libos.create_pipe(&format!("{}:tx", name)) {
-        Ok(qd) => qd,
-        Err(e) => {
-            // Clean up open pipes on error.
-            close(&mut libos, pipeqd_rx);
-            anyhow::bail!("failed to open memory queue: {:?}", e)
-        },
-    };
+        // Setup the Tx pipe.
+        self.tx_pipeqd = match self.libos.create_pipe(&format!("{}:tx", pipe_name)) {
+            Ok(qd) => Some(qd),
+            Err(e) => {
+                if let Some(rx_pipeqd) = self.rx_pipeqd {
+                    close(&mut self.libos, rx_pipeqd);
+                    self.rx_pipeqd = None;
+                }
+                anyhow::bail!("failed to open memory queue: {:?}", e)
+            },
+        };
 
-    // Send and receive bytes in a locked step.
-    for i in 0..NROUNDS {
-        // Clean up open pipes on error.
-        if let Err(e) = pop_and_wait(&mut libos, pipeqd_rx, BUFFER_SIZE, i) {
-            close(&mut libos, pipeqd_rx);
-            close(&mut libos, pipeqd_tx);
-            anyhow::bail!("failed to pop memory queue: {:?}", e);
+        // Send and receive bytes in a locked step.
+        for i in 0..NROUNDS {
+            if let Err(e) = pop_and_wait(
+                &mut self.libos,
+                self.rx_pipeqd.expect("should be a valid pipe qd"),
+                BUFFER_SIZE,
+                i,
+            ) {
+                anyhow::bail!("failed to pop memory queue: {:?}", e);
+            }
+
+            if let Err(e) = push_and_wait(
+                &mut self.libos,
+                self.tx_pipeqd.expect("should be a valid pipe qd"),
+                BUFFER_SIZE,
+                i,
+            ) {
+                anyhow::bail!("failed to push memory queue: {:?}", e);
+            }
+            println!("pong {:?}", i);
         }
 
-        if let Err(e) = push_and_wait(&mut libos, pipeqd_tx, BUFFER_SIZE, i) {
-            close(&mut libos, pipeqd_rx);
-            close(&mut libos, pipeqd_tx);
-            anyhow::bail!("failed to push memory queue: {:?}", e);
-        }
-        println!("pong {:?}", i);
+        Ok(())
     }
+}
 
-    close(&mut libos, pipeqd_rx);
-    close(&mut libos, pipeqd_tx);
+// Drop implementation for the pipe server.
+impl Drop for PipeServer {
+    fn drop(&mut self) {
+        if let Some(rx_pipeqd) = self.rx_pipeqd {
+            close(&mut self.libos, rx_pipeqd);
+        }
+        if let Some(tx_pipeqd) = self.tx_pipeqd {
+            close(&mut self.libos, tx_pipeqd);
+        }
+    }
+}
 
-    Ok(())
+// The pipe client.
+pub struct PipeClient {
+    /// Underlying libOS.
+    libos: LibOS,
+    /// Tx pipe qd.
+    tx_pipeqd: Option<QDesc>,
+    /// Rx pipe qd.
+    rx_pipeqd: Option<QDesc>,
```

Comment:
Could you rollback changes applied to the name of these variables?

Reply:
I changed the names to make them easier to read. The difference in the name is at the very beginning now, so they stand out and reduce ambiguity while reading the code. Do let me know if you think differently.

Reply:
I have a different perspective on this. Here is the rationale:

- The code base uses the previous name convention. Checkout mainline code for Catloop.
- This PR does not rename variables repository-wide.
- We end up having different naming conventions for this variable.
- Having the new naming style being easier/harder is something personal.

Reply:
Done. Agreed, it's better to have consistent naming style across, so I have reverted the name to the earlier form.

About the last point, names being easier to read: it's a general/common practice to have similar variable names differ as early as possible (possibly first char) which leads to faster code reading and comprehension. Though this doesn't matter in this context thought because the code base is very small and there aren't many variables to be confused about.

Reply:
Oh, I see. Well, I particularly never heard about this common/general practice. If you have pointers for this, let me know.

I follow the [Linux Kernel programming style](https://www.kernel.org/doc/Documentation/process/coding-style.rst), which strongly advocates for having short name for variables.

Reply:
Your point about alignment is valid. If the variables are mostly used together, this alignment does makes it easier to read.

On the other side, Kevlin Henney has talked and documented/presented this practice (will share a link to his talk/doc) and refers to naming tips from the "Clean Code" book in conjunction.

Well, this discussion was not about short/long names, but since you brought it up:
- I'm not an advocate for long/short names, I just want to ensure that names are least ambiguous and most readable
- The "Clean Code" book by Robert C. Martin is an excellent and widely accepted documentation about variable naming
- He mentions that variable names should be proportional to the scope, and other ways to reduce ambiguity
- He also mentions that descriptive names are good if they help in clarifying the context and reducing complexity